### ISSUE TITLE
can not draw accurately

### ISSUE THREAD

**Issue by Charlie441324219** [Victrid/splatplost#4]
![WeChat Image_20210325205725](https://user-images.githubusercontent.com/22875442/112542702-dee8dd80-8d82-11eb-91e6-2d7a3adca385.jpg)
 
![WeChat Image_20210326033617](https://user-images.githubusercontent.com/22875442/112573735-e5934700-8dba-11eb-8392-46a755905050.jpg)
-----------------------------------------update------------------------------
I updated the software and used sudo python ./print.py IMAGE.order 0.1. But the problem are getting worse.
![WeChat Image_20210327191942](https://user-images.githubusercontent.com/22875442/112732375-9ad31580-8f07-11eb-84e5-5e5e120de4db.jpg)

**Comment by Victrid** [comment]
This is caused by bluetooth blockage.
Will be fixed recently.

**Comment by Victrid** [comment]
@Charlie441324219 Please check current version with guidances in the `readme.md`.
Use the delay time method first, as the pointer reset method could not solve the problem, but just reduce the severity.

**Comment by Charlie441324219** [comment]
> @Charlie441324219 Please check current version with guidances in the `readme.md`.
> Use the delay time method first, as the pointer reset method could not solve the problem, but just reduce the severity.

Thx for your response. I used the delay time method, but the problem are getting worse. And can't reopen an issue which repo owner closed it.
 
![WeChat Image_20210327191942](https://user-images.githubusercontent.com/22875442/112734059-6d8b6500-8f11-11eb-895c-40cbf2bf8099.jpg)

**Comment by DaYu525** [comment]
> > @Charlie441324219 Please check current version with guidances in the `readme.md`.
> > Use the delay time method first, as the pointer reset method could not solve the problem, but just reduce the severity.
> 
> Thx for your response. I used the delay time method, but the problem are getting worse. And can't reopen an issue which repo owner closed it.
> 
> ![WeChat Image_20210327191942](https://user-images.githubusercontent.com/22875442/112734059-6d8b6500-8f11-11eb-895c-40cbf2bf8099.jpg)

Same problem, use delay time method too.

**Comment by Victrid** [comment]
That's strange...
I've added a `test_print.py` which uses `matplotlib` pip package to plot. Please use `python ./test_print.py <IMAGE.order>` to check if the printing instructions are incorrect.
Do you use a virtual machine to plot? I know some vm will do "optimization" on IO and mess up all these time controlling things.

**Comment by DaYu525** [comment]
I have tried the delay time 0.5 ，cost 5 hour and same problem.  I will try the new way test_print later and use physical Linux machine in usb device.

**Comment by Victrid** [comment]
Sorry for your wasted time. 
Here's a plot for test, please help downloading and plotting it **without any optimization arguments**, and show me the output for faster bug locating.
This picture is simple, and will take you no more than 10 minutes to plot.
[112743279-696e4e80-8fc8-11eb-9f77-db35d7d39a9b.zip](https://github.com/Victrid/splatplost/files/6216899/112743279-696e4e80-8fc8-11eb-9f77-db35d7d39a9b.zip)

**Comment by DaYu525** [comment]
> Sorry for your wasted time.
> Here's a plot for test, please help downloading and plotting it **without any optimization arguments**, and show me the output for faster bug locating.
> This picture is simple, and will take you no more than 10 minutes to plot.
> [112743279-696e4e80-8fc8-11eb-9f77-db35d7d39a9b.zip](https://github.com/Victrid/splatplost/files/6216899/112743279-696e4e80-8fc8-11eb-9f77-db35d7d39a9b.zip)

ok, im trying now, edit this reply when it done.

**Comment by Victrid** [comment]
Seems like some keys are stuck in `pressed` state.
You may try it with some live-cd if you're using virtual machine.

**Comment by DaYu525** [comment]
I‘ll test later. I forced shotdown my ubuntu in usb device, it can't start now and stuck in the logo(five point below it), I'm trying to repair it.

**Comment by DaYu525** [comment]
I try the test image and my own, looks better now.
Completely accurate in the test pictures.
Little offset in my own.
![WechatIMG149](https://tva1.sinaimg.cn/large/008eGmZEgy1gpa54fqrusj30zk0k0n04.jpg)
![SplatCoat](https://tva1.sinaimg.cn/large/008eGmZEgy1gpa5518ou6j308w03c3zl.jpg)

**Comment by SuperOkazaki** [comment]
It being kind of relevant to this issue, I guess I'll ask here-- is there a recommended "safe" speed for Delay and Key Press ms timing? I'm using the 0.2.0.dev.0 version and having accuracy issues with larger prints. I have my Delay and Key Press set to 200ms each but I still get breaks and lines. What is the recommended setting for this? I don't mind the print taking a long time. 

Also-- is there a breakdown of what "Stable Mode" does? I have tried printing with it, but it didn't seem to make a difference.

**Comment by Victrid** [comment]
@SuperOkazaki Stable mode will send the command of drawing a block as a pile for backend to process, instead of sending a press command every time a press command completes. This may help as the plotting will have less breaks (which is then called stable).

There is no "safe" speed, as switch controllers works on some internal timer (seen from nxbt would be 120Hz? I haven't figured out how this is implemented). It's like a rhythm game, neither early nor later press is good. 80ms is based on an old issue which says its a good tempo. I'm still digging on whether it can be sent based on 120Hz basis rather than millisecond basis.

**Comment by SuperOkazaki** [comment]
> There is no "safe" speed, as switch controllers works on some internal timer (seen from nxbt would be 120Hz? I haven't figured out how this is implemented). It's like a rhythm game, neither early nor later press is good.

makes sense, thanks! I'll keep playing around with it to see what works for me
